src/humanloop/otel/exporter.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import typing
+from queue import Queue
+from threading import Thread
+from typing import Any, Optional
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+
+from humanloop.otel.constants import HL_FILE_OT_KEY, HL_LOG_OT_KEY, HL_OT_EMPTY_VALUE, HL_TRACE_METADATA_KEY
+from humanloop.otel.helpers import is_humanloop_span, read_from_opentelemetry_span
+from humanloop.requests.flow_kernel_request import FlowKernelRequestParams
+from humanloop.requests.prompt_kernel_request import PromptKernelRequestParams
+
+if typing.TYPE_CHECKING:
+    from humanloop.base_client import BaseHumanloop
+
+
+logger = logging.getLogger("humanloop.sdk")
+
+
+class HumanloopSpanExporter(SpanExporter):
+    """Upload Spans created by SDK decorators to Humanloop.
```

Comment:
What happens if something goes wrong with the logging to Humanloop?

We should have retry logic on certain error codes i think

Reply:
E.g. 500s?

Reply:
added logging for errros and logic in case a parent in trace fails